egs/librimix/ConvTasNet/eval.py

```diff
@@ -0,0 +1,149 @@
+from model import make_model_and_optimizer
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--test_dir', type=str, required=True,
+                    help='Test directory including the json files')
+parser.add_argument('--use_gpu', type=int, default=0,
+                    help='Whether to use the GPU for model execution')
+parser.add_argument('--exp_dir', default='exp/tmp',
+                    help='Experiment root')
+parser.add_argument('--n_save_ex', type=int, default=10,
+                    help='Number of audio examples to save, -1 means all')
+
+compute_metrics = ['si_sdr', 'mir_eval_sdr', 'mir_eval_sir', 'mir_eval_sar',
+                   'stoi']
+
+
+def main(conf):
+    # Make the model
+    model, _ = make_model_and_optimizer(conf['train_conf'])
+    # Load best model
+    with open(os.path.join(conf['exp_dir'], 'best_k_models.json'), "r") as f:
+        best_k = json.load(f)
+    best_model_path = min(best_k, key=best_k.get)
+    # Load checkpoint
+    checkpoint = torch.load(best_model_path, map_location='cpu')
+    state = checkpoint['state_dict']
+    state_copy = state.copy()
+    # Remove unwanted keys
+    for keys, values in state.items():
+        if keys.startswith('loss'):
+            del state_copy[keys]
+            print(keys)
+    model = load_state_dict_in(state_copy, model)
+
+    # Handle device placement
+    if conf['use_gpu']:
+        model.cuda()
+    model_device = next(model.parameters()).device
+
+    test_set = LibriMix(conf['test_dir'], None,
+                        conf['sample_rate'],
+                        conf['train_conf']['data']['n_src'])
+
+    loss_func = PITLossWrapper(pairwise_neg_sisdr, mode='pairwise')
+
+    # Randomly choose the indexes of sentences to save.
+    ex_save_dir = os.path.join(conf['exp_dir'], 'examples_mss_8K/')
+    if conf['n_save_ex'] == -1:
+        conf['n_save_ex'] = len(test_set)
+    save_idx = random.sample(range(len(test_set)), conf['n_save_ex'])
+    series_list = []
+    torch.no_grad().__enter__()
+    for idx in tqdm(range(len(test_set))):
+        # Forward the network on the mixture.
+        mix, sources = tensors_to_device(test_set[idx], device=model_device)
+
+        est_sources = model(mix.unsqueeze(0))
+
+        loss, reordered_sources = loss_func(est_sources, sources[None],
+                                            return_est=True)
+
+        mix_np = mix.cpu().data.numpy()
+        sources_np = sources.squeeze().cpu().data.numpy()
+        est_sources_np = reordered_sources.squeeze().cpu().data.numpy()
+        est_sources_np[0, :] = est_sources_np[0, :] / \
+                               np.max(est_sources_np, axis=1)[0] * \
+                               np.max(sources_np, axis=1)[0]
+        est_sources_np[1, :] = est_sources_np[1, :] / \
+                               np.max(est_sources_np, axis=1)[1] * \
+                               np.max(sources_np, axis=1)[1]
```

Comment:
We'll need to get rid of this. Estimate normalization cannot depend directly on the true source.

Reply:
Yeah that's bad. How do you usually deal with normalization ? I guess I could just avoid the clipping by dividing by 'max(est_source)'. Is relu better than sigmoid for mask activation ?

Reply:
This is one way, `src * max(abs(mix))/max(abs(src))` is also used but is not perfect. We could think about better. 
For the mask activation, I remember seeing better results for ReLU but cannot remember where. But, if better, it is marginal.